### PR TITLE
Fixed TxFlowTask and added the fake network to test it

### PR DESCRIPTION
**Pull request by nearmax**

### CODE REVIEW

File: core/txflow/spec/main.tex [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+\section*{Basic notation}
+We say that message $c$ directly approves $p$ if $p$ contains the hash of $c$. We say that $c$ approves $p$, denoted as $p\leftarrow c$ if there is $p'$ such that $c$ directly approves $p'$ and $p'$ approves $p$, i.e. it is a transitive closure. We also allow it to be reflexive so $c\leftarrow c$ is always true.
+
+The domain of the message $c$ is all messages that it approves:
+\[
+\begin{aligned}
+\overline{D}(c) \buildrel\triangle\over = \{p|p\leftarrow c\} \\
```
Comment (mikhailOK) [inline core/txflow/spec/main.tex]:
```suggestion
\overline{D}(c) \triangleq \{p|p\leftarrow c\} \\
```
for a smaller triangle

File: core/txflow/spec/main.tex [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+\begin{document}
+
+\maketitle
+
+TxFlow algorithm is all about building a DAG of messages and recognizing when a certain message has some special properties. The properties of each message depend exclusively on its sub-DAG. We constantly iterate on the improvements of the algorithm in several languages that use different algorithmic approaches, e.g. Rust implementation tries to avoid DAG traversals by precomputing the needed information in each message, while Javascript aims for the simplicity of the implementation. While we do these iterations it easy to lose track of the actual algorithm in the implementation details. This notebook lists the predicates that formally define the properties of the messages and serves as the \textbf{source of truth} for TxFlow.
+
+\section*{Basic notation}
+We say that message $c$ directly approves $p$ if $p$ contains the hash of $c$. We say that $c$ approves $p$, denoted as $p\leftarrow c$ if there is $p'$ such that $c$ directly approves $p'$ and $p'$ approves $p$, i.e. it is a transitive closure. We also allow it to be reflexive so $c\leftarrow c$ is always true.
+
+The domain of the message $c$ is all messages that it approves:
```
Comment (mikhailOK) [inline core/txflow/spec/main.tex]:
```suggestion
The domain of a message $c$ is all messages that it approves:
```

File: core/txflow/spec/main.tex [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+\[m_w^e \text{ -- message with epoch $e$, created by $w$;} \]
+\[L(e) \text{ -- a witness of epoch $e$;} \]
+\[S(c, p) \text{ -- message $c$ contains signature for message $p$, where $p\leftarrow c$;}\]
+
+\section*{Definitions}
+\subsection*{Representative message}
+
+\[
+\begin{aligned}
+R(m_w^x, y) \buildrel\triangle\over =& \text{   } &\textit{Message $m_w^x$ is the representative of epoch $y$ if: } \\
```
Comment (mikhailOK) [inline core/txflow/spec/main.tex]:
```suggestion
R(m_w^x, y) \buildrel\triangle\over =& \text{   } &\textit{Message $m_w^x$ is a representative of epoch $y$ if: } \\
```

File: core/txflow/spec/main.tex [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+\begin{aligned}
+\overline{D}(c) \buildrel\triangle\over = \{p|p\leftarrow c\} \\
+D(c) \buildrel\triangle\over = \overline{D}(c)\setminus\{c\}
+\end{aligned}
+\]
+
+When it matters we annotate the message with the witness that created it and its epoch:
+
+\[m_w^e \text{ -- message with epoch $e$, created by $w$;} \]
+\[L(e) \text{ -- a witness of epoch $e$;} \]
```
Comment (mikhailOK) [inline core/txflow/spec/main.tex]:
```suggestion
\[L(e) \text{ -- the leader of epoch $e$;} \]
```

File: core/txflow/spec/main.tex (hunk 1 above)
Comment (nearmax) [reply core/txflow/spec/main.tex]: Fixed for the rest of the doc.

File: core/txflow/spec/main.tex [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+\title{TxFlow Functional Form}
+\author{NearProtocol}
+\date{Last Updated November 27, 2018}
+
+\begin{document}
+
+\maketitle
+
+TxFlow algorithm is all about building a DAG of messages and recognizing when a certain message has some special properties. The properties of each message depend exclusively on its sub-DAG. We constantly iterate on the improvements of the algorithm in several languages that use different algorithmic approaches, e.g. Rust implementation tries to avoid DAG traversals by precomputing the needed information in each message, while Javascript aims for the simplicity of the implementation. While we do these iterations it easy to lose track of the actual algorithm in the implementation details. This notebook lists the predicates that formally define the properties of the messages and serves as the \textbf{source of truth} for TxFlow.
```
Comment (mfornet) [inline core/txflow/spec/main.tex]: `While we do these iterations it IS easy to lose track...`

Comment (mfornet) [reply core/txflow/spec/main.tex]: I meant there is a missing "IS"

**Pull request merged**